### PR TITLE
Buildmode/spawn QoL update

### PR DESCRIPTION
**Pull request by unid15**
* Spawn and create-datum can automatically change the freshly spawned object's variables if you append all changes at the end of the input string like so:
spawn "/mob/living/carbon/human{name = "Red Dude";color="#FF0000"}"
spawn "/obj/item/stack/sheet/metal{amount=50}"
spawn "/obj/item.{icon_state="hammer";name="Hammer";force=20}"

* Fixed some bugs with buildmode:
  * Fixed the GUI buttons being super sensitive to mouse clicks
  * Fixed weird behaviour when setting a variable to 0
  * Fixed being unable to reset things to their initial appearance sometimes
* In buildmode's edit variable mode, use the middle mouse button to copy the variable's value from an object (also works for appearances).

qol_of_life

:cl:
 * experiment: The "spawn" and "create-datum" admin commands can change the new object's variables automatically after spawning it. Append any variable changes at the end of the string in curly brackets like so: spawn "/obj/item.{name="Item" ; w_class=5}"
 * bugfix: Fixed some bugs with buildmode's edit mode (fixed strange behaviour when setting values to 0, fixed inability to reset object appearances). While in edit mode, middle-click any object to copy its variable.

### CODE REVIEW

**Comment by SonixApache** [comment]
Did you document this anywhere (that isn't this github page)?

**Comment by PJB3005** [comment]
SDQL vs other thing war now.

**Comment by unid15** [comment]
Buildmode var copying is documented in the help button, the spawn feature is documented in the verb description

File: code/WorkInProgress/buildmode.dm [hunk 1]
```diff
@@ -204,8 +205,9 @@ obj/effect/bmode/buildholder/New()
 				if(edit_variable != "appearance") //Special case for appearance
 					master.buildmode.valueholder = variable_set(usr)
 	return 1
-/obj/effect/bmode/buildmode/DblClick(object,location,control,params)
-	return Click(object,location,control,params)
+
+///obj/effect/bmode/buildmode/DblClick(object,location,control,params)
```
Comment (Shadowmech88) [inline code/WorkInProgress/buildmode.dm]: Why comment this out rather than remove it altogether?

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -1292,6 +1302,8 @@ var/global/floorIsLava = 0
 		if(!chosen)
 			return
 
+	_preloader = new(varchanges, chosen)
```
Comment (PJB3005) [inline code/modules/admin/admin.dm]: Uh, you're not doing anything with this, are you?

Comment (unid15) [reply code/modules/admin/admin.dm]: Preloader is always deleted after processing one atom, so there won't be any problem with that
I'll clarify that in a comment though

Comment (unid15) [reply code/modules/admin/admin.dm]: oh you probably meant something else

_preloader is hooked to atom/New(), so there's no need to call anything after initializing it

**Pull request merged**